src/go/intelligence/pipeline/add_annotations.go

```diff
@@ -84,6 +141,26 @@ func makeHTMLhref(url string) string {
 	return "<a target=_blank href=" + url + ">URL</a>"
 }
 
+//getTagsIntersection for finding intersection between two list of dashboard tags.
```

Comment:
the description and name of this function is misleading. It does not find the intersection, since the intersection is a list of something, instead what it does it finds if entries from one list appear in another list. If so, the implementation is not correct either since you can simply loop over one list and find if its entry exists in another, if so you can return immediately a true boolean.

Please clarify the function behavior, rename it properly and simply.

Reply:
Valentin,
It is actually finding the intersection between two lists, but what I am returning is if the resultant list after getting an intersection is empty or not. So, intuitively I am checking if there are any dashboards common between all dashboards and dashboards specific to an annotation.

```

//getTagsIntersection for finding intersection between two list of dashboard tags.
func getTagsIntersection(annotationDashboardTags, allDashboardsTags []string) bool {

	var intersectionResult []string      ------>>>> HERE THE INTERSECTION RESULT IS GETTING STORED

	hashMap := make(map[string]bool) //a hashmap which helps to reduce the intersection operation from O(n^2) to O(n).

	for _, val := range allDashboardsTags {
		hashMap[val] = true
	}

	for _, each := range annotationDashboardTags {
		if _, ok := hashMap[each]; ok {
			intersectionResult = append(intersectionResult, each)
		}
	}

	return len(intersectionResult) > 0            ----> CHECKING IF THE RESULTANT LIST AFTER INTERSECTION IS NOT EMPTY
}
```

Reply:
I used intersection concept using hashmap to optimize the process from O(n^2) to O(n)

Reply:
I have made a few changes in description now.